File: server/Models/GiftModel.cs
namespace server.Models
{
    public class GiftModel
    {
        public int Id { get; set; }
        public string Description { get; set; } = string.Empty;
        public int CategoryId { get; set; }
        public decimal Price { get; set; }
        public int DonorId { get; set; }
        public UserModel Donor { get; set; } = default!;
        public CategoryModel Category { get; set; } = default!;
        public string ImageUrl { get; set; } = string.Empty;
        public Boolean HasWinning { get; set; } = false;

       
    }
}

File: client/src/app/core/models/gift-model.ts

export interface GiftResponseDto {
  id: number;
  description: string;
  price: number;
  categoryName: string;
  categoryId: number;
  donorId: number;
  imageUrl: string;
}

export interface GiftCreateDto {
  description: string;
  price: number;
  categoryId: number;
  donorId: number;
  imageUrl?: string;
}

export interface GiftUpdateDto {
  description?: string;
  price?: number;
  categoryId?: number;
  imageUrl?: string;
}

export interface GiftStatsDto {
  giftId: number
  description: string,
  ticketsSold: number,
  uniqueBuyers: number,
  hasWinning: boolean
}

export enum PriceSort {
  None = 'None',
  Ascending = 'Ascending',
  Descending = 'Descending',
}



